### PR TITLE
test: Use universal_mocks in each unit test

### PR DESCRIPTION
**Pull request by ktrzcinx**
test: Use universal_mocks in each unit test

In unit tests there are a lot of mocks duplication for mocks related
with memory management. Universal mock should be solution for this,
problem and contains implementation for them.

Added linking universal_mock to each test by default.

Marked each mock as WEAK to make it possible to overwrite this
implementation in specific test.

To use assert_false in __panic function, added dependence of
universal_mock from cmock.

Signed-off-by: Karol Trzcinski <karolx.trzcinski@linux.intel.com>

### CODE REVIEW

**Comment by jajanusz** [comment]
Please btw change name of these mocks to common, universal is misleading here

File: test/cmocka/src/common_mocks.c [hunk 1]
```diff
@@ -24,3 +65,13 @@ int memcpy_s(void *dest, size_t dest_size,
...
 	return 0;
 }
+
+void WEAK __panic(uint32_t p, char *filename, uint32_t linenum)
+{
+	(void)p;
+	(void)filename;
+	(void)linenum;
+
+	print_error("panic: %s:%d (code %d)", filename, linenum, p);
```
Comment (paulstelian97) [inline test/cmocka/src/common_mocks.c]: print_error can be defined out? If not, the casts to void above aren't needed.

Comment (ktrzcinx) [reply test/cmocka/src/common_mocks.c]: yes, casting is redundant.

File: test/cmocka/src/common_mocks.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Jakub Dabek <jakub.dabek@linux.intel.com>
```
Comment (jajanusz) [inline test/cmocka/src/common_mocks.c]: @ktrzcinx Please add yourself as 2nd author

Comment (paulstelian97) [reply test/cmocka/src/common_mocks.c]: Also copyright 2019-2020?

Comment (ktrzcinx) [reply test/cmocka/src/common_mocks.c]: Thanks, done

**Comment by jajanusz** [comment]
> CI says it's all good, but lets wait for more approvals.

There are 6 checks passed, non of them even checks unit tests, please wait for quickbuild (there are 8 checks in total).
@lgirdwood Maybe we should mark these CIs (travis/jenkins/qb) as required so they are always shown in CIs list on GH, even when CI haven't queue them yet. Our kernel fork has it for jenkins.

File: test/cmocka/src/common_mocks.c [hunk 3]
```diff
@@ -1,14 +1,56 @@
 // SPDX-License-Identifier: BSD-3-Clause
 //
-// Copyright(c) 2019 Intel Corporation. All rights reserved.
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
```
Comment (jajanusz) [inline test/cmocka/src/common_mocks.c]: The date should be the first time it was done (2019) or range when it was done and edited (2019-2020)

**Comment by lgirdwood** [comment]
> > CI says it's all good, but lets wait for more approvals.
> 
> There are 6 checks passed, non of them even checks unit tests, please wait for quickbuild (there are 8 checks in total).
> @lgirdwood Maybe we should mark these CIs (travis/jenkins/qb) as required so they are always shown in CIs list on GH, even when CI haven't queue them yet. Our kernel fork has it for jenkins.

Good point, I heard from @zrombel that QB does not even show as pending until it starts running. @zrombel any issues if we mark these as required ?

File: test/cmocka/src/common_mocks.c (hunk 3 above)
Comment (ktrzcinx) [reply test/cmocka/src/common_mocks.c]: Ok, done

**Comment by zrombel** [comment]
@lgirdwood yes, **pening** is shown after the build is started. I don't think there should be any issues if QB is marked as required, but it's more github thing so I'm not 100% sure.

**Comment by jajanusz** [comment]
@lgirdwood please approve if you are ok with it.
BTW https://help.github.com/en/github/administering-a-repository/enabling-required-status-checks, can you make `Internal Intel CI System/merge/build` required? Jenkins and Travis show almost instantly anyway so we don't need them in the list.

**Comment by lgirdwood** [comment]
CI usual suspect.

**Pull request merged**